src/QbXml/Objects/Types/DATETIMETYPE.cs

```diff
@@ -156,42 +270,167 @@ public override int GetHashCode()
             return default(DateTime);
         }
 
+
+        public static bool operator <(DATETIMETYPE a, DATETIMETYPE b)
+        {
+            return a.CompareTo(b) < 0;
+        }
+
+        public static bool operator >(DATETIMETYPE a, DATETIMETYPE b)
+        {
+            return a.CompareTo(b) > 0;
+        }
+
         public int CompareTo(DATETIMETYPE other)
         {
-            return this.value.CompareTo(other.value);
+            if (other == null)
+            {
+                return 1;
+            }
+
+            if (_ignoreOffset || other._ignoreOffset)
+            {
+                return _value.DateTime.CompareTo(other._value.DateTime);
+            }
+
+            return _value.CompareTo(other._value);
         }
 
         public System.Xml.Schema.XmlSchema GetSchema()
         {
             return null;
         }
 
+        [Obsolete]
         public void ReadXml(System.Xml.XmlReader reader)
         {
+            if (!_canReadXml)
+            {
+                // This class needs to be immutable as possible
+                // Only allow this when constructed using the empty constructor to only be used by deserialization
+                throw new InvalidOperationException("This method not to be used by user code");
+            }
+
+            // Only allow ReadXML once to prevent user code from calling this method after deserialized
+            _canReadXml = false;
+
             reader.MoveToContent();
             var isEmptyElement = reader.IsEmptyElement;
             reader.ReadStartElement();
-            if (!isEmptyElement)
+
+            if (isEmptyElement)
+            {
+                _value = new DateTimeOffset(1970, 1, 1, 0, 0, 0, 0, TimeSpan.Zero);
+                _ignoreOffset = true;
+                return;
+            }
+
+            var str = reader.ReadContentAsString();
+            var timeZone = QbXmlResponse.qbXmlResponseOptionsStatic?.TimeZoneBugFix;
+
+            if (timeZone == null && str != null && str.Length > 19)
+            {
+                // QuickBooks has inaccurate offset values
+                // The DateTime is otherwise accurate (follows DST), but the offset does not follow DST
+                // To accomodate for this, simply ignore the offset portion of the string, but only when parsing from XML
+                str = str.Substring(0, 19);
```

Comment:
Hey, by doing this, don't you assume the data coming from QuickBooks will be AsUniversal if the timezone is not specified?
I know that if you don't set the timezone properly, you mess up your DST.
But now, let's say you don't setup your timezone, you will get all the dates as UTC?

Reply:
DST only gets messed up when you use `DateTime.Parse()` with the offset from QuickBooks included in the string. By dropping the offset before parsing, we avoid this problem, and is the basis for my change to handle dates even without a `TimeZoneBugFix` set. It won't be UTC, it will be Unspecified.

Did I understand your question correctly?

Reply:
Ah, gotcha. Yes, it will work fine. I tested this scenario. You have to keep the ToDateTime() as Unspecified though. ToUniversalTime() will break it. I'll make a code example to show this, and perhaps you could do a test in your own environment just to double check me.

Reply:
But wouldn't you want to save in universal time?

Reply:
Before, you could just subtract/add the offset to get your utc. Now do we completely lose that info?

Reply:
> But wouldn't you want to save in universal time?

That _would_ be nice yes. But we don't have to, especially because due to this QuickBooks bug, we don't get that luxury unless we make the developer get an _accurate_ time zone for their QuickBooks install. 

Don't have that? No big deal, we'll just store the date without any time zone, and treat it as the unknown local time of the QuickBooks machine all the way through. Basically we need to consider it like NodaTime's [LocalDateTime](https://nodatime.org/2.4.x/api/NodaTime.LocalDateTime.html)

This all works fine, no matter the time zone difference between QuickBooks and server, and whether or not either machine is in a DST enabled zone, because QuickBooks allows us to query date ranges without the offset included on the date. 

How about a more practical example... let's say the newest customer returned from a sync has the following in the response xml:
```
...<TimeModified>2019-02-20T10:36:51-06:00</TimeModified>...
```
So QuickBooks is saying the offset is -06:00, however that doesn't tell us what zone they are in, nor if that zone has DST, and due to the bug we can't trust it. But that's ok... we can do searches without that information since QuickBooks lets us. Let's instead _pretend_ quickbooks sent us the following instead:
```
...<TimeModified>2019-02-20T10:36:51</TimeModified>...
```
So we'll store that value in a database, or however the developer feels as the latest modified date for a customer that has been synced. Next sync operation we want to return every customer that has been modified since then, so we want the request xml to include the following:
```
<CustomerQueryRq metaData="ENUMTYPE" iterator="ENUMTYPE" iteratorID="UUIDTYPE">
    // ...
    <FromModifiedDate>2019-02-20T10:36:51</FromModifiedDate>
    // ...
</CustomerQueryRq>
```
With my tests, we miss zero data doing this, which matches what QuickBooks documentation says, and we never had to convert to UTC, or even know the offset. Simple :) 

And that is why, when a `TimeZoneBugFix` is not set, I "pretend" QuickBooks never even sent us an offset by doing the `str = str.Substring(0, 19);` you are commenting on.

Reply:
OK, I think we should add this in the documentation then. If you don't set your timezone, then you should save in your DB "2019-02-20T10:36:51" so that you can requery the exact "2019-02-20T10:36:51".
I'm not sure what people use out there if they use their TimeZone, but I just wanted to make sure we don't completely mess up.

Reply:
Good idea. 

And note... a lot of my changes were geared around helping developers NOT mess this up if they didn't know better. Check out my tests... I know there are a lot of them, but they may illustrate the case. The developer can do any of the following, save the result, and re-parse directly to DateTime or DATETIMETYPE and my above described behavior will work:
```
DATETIMETYPE.ToString();
DATETIMETYPE.ToDateTime() // Could be stored in MS SQL as DateTime2
DATETIMETYPE.ToDateTime().ToString() 
```
Doing any of the following will cause problems. No way to prevent this as it is just stupid problems with `System.DateTime`, and why encouraging NodaTime's LocalDateTime would be nice
```
DATETIMETYPE.ToDateTime().ToUniversalTime();
DATETIMETYPE.ToDateTime().ToLocalTime();
```

Reply:
Sounds good. I'll be waiting for your next commit to handle the point 1 and 2. Then we can update the rest in different PR.
Do you mind to join a chat room just if we need to talk more about this or knowing if I can help in other spots. I sent you an invite on LinkedIn. I can share more info from there.

Reply:
I just pushed 2 more tests that highlight handling of various round-trip scenarios. Each assert should probably be its own test, but it was much faster to write this way.